superset-frontend/src/explore/components/DataTablesPane/index.tsx

```diff
@@ -192,7 +250,25 @@ export const DataTablesPane = ({
               [resultType]: json.result[0].data,
             }));
           }
-
+          const checkCols = json?.result[0]?.data?.length
+            ? Object.keys(json.result[0].data[0])
+            : null;
+          if (!json.result[0].colnames && checkCols?.length) {
+            let tabKey: string = RESULT_TYPES.results;
+            if (resultType === RESULT_TYPES.samples) {
+              tabKey = RESULT_TYPES.samples;
+            }
+            setColumnNames({
```

Comment:
We should use function syntax when we update state using the old state
```setColumnNames(prevColumnNames => ({
    ...prevColumnNames,
    [tabKey]: checkCols,
}));
```
https://reactjs.org/docs/hooks-reference.html#functional-updates